### PR TITLE
add verifiable load contract

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: core/scripts/chaincli/command/keeper/verifiable_load.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+	"github.com/smartcontractkit/chainlink/core/scripts/chaincli/config"
+	"github.com/smartcontractkit/chainlink/core/scripts/chaincli/handler"
+)
+
+// verifiableLoad represents the command to get verifiable load testing details
+var verifiableLoad = &cobra.Command{
+	Use:   "verifiable-load",
+	Short: "Print verifiable load testing details to console",
+	Long:  `Print verifiable load testing details to console, including details of every active upkeeps and total result`,
```
Comment (ferglor) [inline core/scripts/chaincli/command/keeper/verifiable_load.go:14]: `Print verifiable load testing details to console, including details of every active upkeep and total results` 🏄

**Comment by FelixFan1992** [comment]
This PR:
1. refactors the Verifiable Load Contract for conditional upkeeps and mercury upkeeps.
2. provides a go script in chain CLI to read VLC results reliably.
3. generates go wrappers for VLC and updates `deployUpkeeps` function to deploy them.

File: core/scripts/chaincli/handler/keeper_verifiable_load.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	defer wg.Done()
+
+	var bucketDelays []*big.Int
+	var err error
+	if getTimestampBucket {
+		for i := 0; i < 3; i++ {
+			bucketDelays, err = v.GetTimestampDelays(opts, id, bucketNum)
+			if err != nil {
+				log.Printf("failed to get timestamp bucketed delays for upkeep id %s timestamp bucket %d: %v, retrying...", id.String(), bucketNum, err)
+				time.Sleep(time.Second * 3)
```
Comment (ferglor) [inline core/scripts/chaincli/handler/keeper_verifiable_load.go:192]: This retry delay could also be a const?

**Review by ferglor — APPROVED** [review]
Just a few comments but only one or two that need addressed, looks good! 🙌

File: core/scripts/chaincli/command/keeper/verifiable_load.go (hunk 1 above)
Comment (FelixFan1992) [reply core/scripts/chaincli/command/keeper/verifiable_load.go:14]: done

File: core/scripts/chaincli/handler/keeper_verifiable_load.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	DelayTimestampBuckets map[uint16][]float64
+	SortedAllDelays       []float64
+	TotalDelayBlock       float64
+	TotalPerforms         uint64
+}
+
+func (ui *UpkeepInfo) AddBucket(bucketNum uint16, bucketDelays []float64) {
+	ui.mu.Lock()
+	ui.DelayBuckets[bucketNum] = bucketDelays
+	ui.mu.Unlock()
```
[the remark this answers is not in the document]
Comment (FelixFan1992) [reply core/scripts/chaincli/handler/keeper_verifiable_load.go:33]: done.

File: core/scripts/chaincli/handler/keeper_verifiable_load.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+			Bucket:                b,
+			TotalPerforms:         c.Uint64(),
+			DelayBuckets:          map[uint16][]float64{},
+			DelayTimestampBuckets: map[uint16][]float64{},
+		}
+
+		var delays []float64
+		var wg1 sync.WaitGroup
+		for i := uint16(0); i <= b; i++ {
+			wg.Add(1)
```
Comment (ferglor) [inline core/scripts/chaincli/handler/keeper_verifiable_load.go:154]: Oh this is wrong; I think it needs to be `wg1.Add()`?

File: core/scripts/chaincli/handler/keeper_verifiable_load.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+		wg1.Wait()
+
+		// get all the timestamp buckets of an upkeep. performs which happen every 1 hour after the first perform fall into the same bucket.
+		t, err := v.TimestampBuckets(opts, id)
+		if err != nil {
+			log.Fatalf("failed to get timestamp bucket for %s: %v", id.String(), err)
+		}
+		info.TimestampBucket = t
+		for i := uint16(0); i <= t; i++ {
+			wg.Add(1)
```
Comment (ferglor) [inline core/scripts/chaincli/handler/keeper_verifiable_load.go:166]: Same here 👍

File: core/scripts/chaincli/handler/keeper_verifiable_load.go [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+const (
+	// WorkerNum is the total number of workers calculating upkeeps' delay summary
+	WorkerNum = 5
+	// RetryDelay is the time the go routine will wait before calling the same contract function
+	RetryDelay = 1 * time.Second
+	// RetryNum defines how many times the go routine will attempt the same contract call
+	RetryNum = 3
+	// MaxUpkeepNum defines the size of channels. Increase if there are lots of upkeeps.
+	MaxUpkeepNum = 100
```
Comment (ferglor) [inline core/scripts/chaincli/handler/keeper_verifiable_load.go:26]: I would have all of these consts as unexported if possible?

**Review by ferglor — CHANGES_REQUESTED** [review]

File: core/scripts/chaincli/handler/keeper_verifiable_load.go (hunk 4 above)
Comment (FelixFan1992) [reply core/scripts/chaincli/handler/keeper_verifiable_load.go:154]: done

File: core/scripts/chaincli/handler/keeper_verifiable_load.go (hunk 5 above)
Comment (FelixFan1992) [reply core/scripts/chaincli/handler/keeper_verifiable_load.go:166]: done

File: core/scripts/chaincli/handler/keeper_verifiable_load.go (hunk 6 above)
Comment (FelixFan1992) [reply core/scripts/chaincli/handler/keeper_verifiable_load.go:26]: done

**Review by ferglor — APPROVED** [review]
Looks good! I'm not sure if we'll need to limit the concurrent `getBucketData` calls, but that's an easy change if we need to in future 👍

**Pull request merged**